### PR TITLE
Release v2.19.5

### PR DESCRIPTION
**Pull request by zhiqiangjin**
### Notes:
- Supported Controller version: **UserConnect-6.4**
- Supported Terraform version: **v1.0**

#### Accounts
1. Implemented support for AWSChina in **aviatrix_account** and data source with the following new attributes:
- ``awschina_role_app``
- ``awschina_role_ec2``
2. Implemented support for AzureChina in **aviatrix_account** and data source with the following new attributes:
- ``azurechina_subscription_id``
- ``azurechina_directory_id``
- ``azurechina_application_id``
- ``azurechina_application_key``
3. Implemented support for AWSTS in **aviatrix_account** and data source with the following new attributes:
- ``awsts_account_number``
- ``awsts_cap_url``
- ``awsts_cap_agency``
- ``awsts_cap_mission``
- ``awsts_cap_role_name``
- ``awsts_cap_cert``
- ``awsts_cap_cert_key``
- ``awsts_ca_chain_cert``
4. Implemented support for AWSS in **aviatrix_account** and data source with the following new attributes:
- ``awss_account_number``
- ``awss_cap_url``
- ``awss_cap_agency``
- ``awss_cap_account_name``
- ``awss_cap_role_name``
- ``awss_cap_cert``
- ``awss_cap_cert_key``
- ``awss_ca_chain_cert``

#### Firewall Network
1. Implemented support for Fail Close and Network List Excluded From East-West Inspection in **aviatrix_firenet**
2. Implemented support for colon character in ``tags`` in **aviatrix_firewall_instance**

#### Gateway
1. Implemented support for Applying Route Entry in **aviatrix_gateway**
2. Implemented support for AWS Top Secret provider in **aviatrix_gateway**
3. Implemented support for AWS Secret provider in **aviatrix_gateway**
4. Implemented support for configuring gateway keepalive settings
- **aviatrix_controller_gateway_keepalive_config**
5. Implemented support for colon character in ``tags`` in **aviatrix_gateway**

#### Multi-Cloud Transit
1. Implemented support for AWS Top Secret provider in **aviatrix_spoke_gateway**
2. Implemented support for AWS Secret provider in **aviatrix_transit_gateway**
3. Implemented support for colon character in ``tags`` in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
4. Implemented support for connection based BGP prepending in **aviatrix_transit_external_device_conn** and **aviatrix_vgw_conn**

#### TGW Orchestrator
1. Implemented support for the following attribute in **aviatrix_aws_tgw_vpn_conn**
- ``enable_global_acceleration``

### Enhancements:
1. Allowed the value "aviatrix" for the attribute ``host_os`` to support managed cloudN deployment
2. Added support for computed attribute``peering_ha_security_group_id`` in **aviatrix_gateway**
3. Added support for computed attributes ``availability_domains`` and ``fault_domains`` in **aviatrix_vpc** and data source
4. Added support for Panorama in **aviatrix_firenet_vendor_integration** data source

### Bug Fixes:
1. Fixed issue where creating, updating or deleting **aviatrix_controller_cert_domain_config** may cause timeout
2. Fixed issue where disabling Egress fails when Egress is enabled without setting Egress Static CIDRs in **aviatrix_firenet**
3. Fixed issue where setting "account_name" will cause panic in **aviatrix_rbac_group_access_account_attachment**
4. Fixed issue where context deadline exceeded error happens in following resources
- **aviatrix/resource_aviatrix_account**
- **aviatrix/resource_aviatrix_aws_tgw_connect**
- **aviatrix/resource_aviatrix_aws_tgw_connect_peer**
- **aviatrix/resource_aviatrix_aws_tgw_intra_domain_inspection**
- **aviatrix/resource_aviatrix_aws_tgw_security_domain**
- **aviatrix/resource_aviatrix_aws_tgw_security_domain_connection**
- **aviatrix/resource_aviatrix_cloudn_transit_gateway_attachment**
- **aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config**
- **aviatrix/resource_aviatrix_controller_cert_domain_config**
- **aviatrix/resource_aviatrix_controller_email_exception_notification_config**
- **aviatrix/resource_aviatrix_copilot_association**
- **aviatrix/resource_aviatrix_gateway_certificate_config**
5. Fixed issue where ``local_subnet_cidr`` can't be updated for a mapped connection in **aviatrix_site2cloud**
6. Fixed issue where updating access account to swap custom IAM roles for gateways fails
7. Fixed issue where updating ``single_az_ha`` does not apply to HA gateway in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
8. Fixed issue where enabling EBS volume encryption after initial gateway deployment only applies to primary gateway in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,89 @@
...
+1. Implemented support for Fail Close and Network List Excluded From East-West Inspection in **aviatrix_firenet**
+2. Implemented support for colon character in ``tags`` in **aviatrix_firewall_instance**
+
+#### Gateway
+1. Implemented support for Applying Route Entry in **aviatrix_gateway**
+2. Implemented support for AWS Top Secret provider in **aviatrix_gateway**
+3. Implemented support for AWS Secret provider in **aviatrix_gateway**
+4. Implemented support for configuring gateway keepalive settings
+- **aviatrix_controller_gateway_keepalive_config**
+5. Implemented support for colon character in ``tags`` in **aviatrix_gateway**
```
Comment (mlin-aviatrix) [inline CHANGELOG.md]: Colons were already supported in tags before this. The API change should not result in any changes from the user's perspective so I'm not sure we need this in the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,89 @@
-## 2.19.4 (Unreleased)
+## 2.19.5 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-6.4**
+- Supported Terraform version: **v1.0**
```
Comment (CyrusJavan) [inline CHANGELOG.md]: Can we say v1.x instead?
```
- Supported Terraform version: **v1.x**
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,89 @@
...
+### Features:
+#### Accounts
+1. Implemented support for AWSChina in **aviatrix_account** and data source with the following new attributes:
+- ``awschina_role_app``
+- ``awschina_role_ec2``
+2. Implemented support for AzureChina in **aviatrix_account** and data source with the following new attributes:
+- ``azurechina_subscription_id``
+- ``azurechina_directory_id``
+- ``azurechina_application_id``
+- ``azurechina_application_key``
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: We already released this in R2.19.0

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,89 @@
...
+- ``awss_cap_cert``
+- ``awss_cap_cert_key``
+- ``awss_ca_chain_cert``
+
+#### Firewall Network
+1. Implemented support for Fail Close and Network List Excluded From East-West Inspection in **aviatrix_firenet**
+2. Implemented support for colon character in ``tags`` in **aviatrix_firewall_instance**
+
+#### Gateway
+1. Implemented support for Applying Route Entry in **aviatrix_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 1. Implemented support for applying route entry in **aviatrix_gateway_dnat**

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,89 @@
...
+2. Implemented support for AWS Top Secret provider in **aviatrix_gateway**
+3. Implemented support for AWS Secret provider in **aviatrix_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Not sure if calling it 'provider' is correct. Would referring to it as support for the 2 new cloud types be more clear?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,89 @@
...
+
+#### TGW Orchestrator
+1. Implemented support for the following attribute in **aviatrix_aws_tgw_vpn_conn**
+- ``enable_global_acceleration``
+
+### Enhancements:
+1. Allowed the value "aviatrix" for the attribute ``host_os`` to support managed cloudN deployment
+2. Added support for computed attribute``peering_ha_security_group_id`` in **aviatrix_gateway**
+3. Added support for computed attributes ``availability_domains`` and ``fault_domains`` in **aviatrix_vpc** and data source
+4. Added support for Panorama in **aviatrix_firenet_vendor_integration** data source
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: This is actually a new data source : **aviatrix_firenet_firewall_manager** to support Panorama setup

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,89 @@
...
+
+### Bug Fixes:
+1. Fixed issue where creating, updating or deleting **aviatrix_controller_cert_domain_config** may cause timeout
+2. Fixed issue where disabling Egress fails when Egress is enabled without setting Egress Static CIDRs in **aviatrix_firenet**
+3. Fixed issue where setting "account_name" will cause panic in **aviatrix_rbac_group_access_account_attachment**
+4. Fixed issue where context deadline exceeded error happens in following resources
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 4. .... in the following resources:

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,89 @@
...
+- **aviatrix/resource_aviatrix_aws_tgw_connect_peer**
+- **aviatrix/resource_aviatrix_aws_tgw_intra_domain_inspection**
+- **aviatrix/resource_aviatrix_aws_tgw_security_domain**
+- **aviatrix/resource_aviatrix_aws_tgw_security_domain_connection**
+- **aviatrix/resource_aviatrix_cloudn_transit_gateway_attachment**
+- **aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config**
+- **aviatrix/resource_aviatrix_controller_cert_domain_config**
+- **aviatrix/resource_aviatrix_controller_email_exception_notification_config**
+- **aviatrix/resource_aviatrix_copilot_association**
+- **aviatrix/resource_aviatrix_gateway_certificate_config**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: I think this list is a typo with "/"s ?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,80 @@
...
+- ``awss_cap_agency``
+- ``awss_cap_account_name``
+- ``awss_cap_role_name``
+- ``awss_cap_cert``
+- ``awss_cap_cert_key``
+- ``awss_ca_chain_cert``
+
+#### Firewall Network
+1. Implemented support for Fail Close and Network List Excluded From East-West Inspection in **aviatrix_firenet**
+2. Implemented support for colon character in ``tags`` in **aviatrix_firewall_instance**
```
Comment (inc1t3Ful) [inline CHANGELOG.md:29]: As Matthew has stated previously, colons are already supported. Perhaps he can provide a better description of the json tag change. This change is supposed to be completely transparent to user workflow

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,80 @@
...
+1. Implemented support for applying route entry in **aviatrix_gateway_dnat**
+2. Implemented support for AWS Top Secret cloud in **aviatrix_gateway**
+3. Implemented support for AWS Secret cloud in **aviatrix_gateway**
+4. Implemented support for configuring gateway keepalive settings
+- **aviatrix_controller_gateway_keepalive_config**
+
+#### Multi-Cloud Transit
+1. Implemented support for AWS Top Secret provider in **aviatrix_spoke_gateway**
+2. Implemented support for AWS Secret provider in **aviatrix_transit_gateway**
+3. Implemented support for colon character in ``tags`` in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md:41]: see previous comment regarding `tags`

**Pull request closed without merging**